### PR TITLE
FC: Search UI Improvements

### PR DESCRIPTION
**Pull request by vardges-stripe**
## Summary

- Removes loading indicator from searching
- Adds pulsing animation to search icon while search is active
- Changes row animations in the table view
## Motivation

Better UX
## Testing

BEFORE:
https://user-images.githubusercontent.com/92807969/216705836-6d57d913-4a0b-404c-b1c4-7852511266af.mov

AFTER:

https://user-images.githubusercontent.com/92807969/216705893-0c44becc-9f2d-45df-bdce-0cab73b44bfa.mov

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchTableView.swift [hunk 1]
```diff
@@ -83,6 +83,8 @@ final class InstitutionSearchTableView: UIView {
             cell.customize(with: institution)
             return cell
         }
+        dataSource.defaultRowAnimation = .fade
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchTableView.swift:86]: nice touch

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchBar.swift [hunk 2]
```diff
@@ -139,6 +150,26 @@ final class InstitutionSearchBar: UIView {
         layer.borderColor = searchBarBorderColor.cgColor
         layer.borderWidth = searchBarBorderWidth
     }
+
+    private var searchIndicatorActive: Bool = false
+
+    func updateSearchingIndicator(_ isSearching: Bool) {
+        guard isSearching != self.searchIndicatorActive else { return }
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchBar.swift:157]: I believe this is so we don't add the animation to the layer multiple times? 

If correct, maybe one way to make this explicit & remove the extra variable would be to do something like:

```

func animateSearchIndicator(_ shouldAnimateSearchIndicator: Bool) {
        if shouldAnimateSearchIndicator {
            let isAnimationActive = textField.layer.animation(forKey: "pulseAnimation") != nil
            if !isAnimationActive {
                let opacityAnimation = CABasicAnimation(keyPath: "opacity")
                opacityAnimation.fromValue = 0.6
                opacityAnimation.toValue = 0.25
                opacityAnimation.repeatCount = 10000
                opacityAnimation.duration = 0.3
                opacityAnimation.autoreverses = true
    
                searchIconView.layer.add(opacityAnimation, forKey: "pulseAnimation")
            }
        } else {
            searchIconView.layer.removeAnimation(forKey: "pulseAnimation")
        }
}
```

**Review by kgaidis-stripe — DISMISSED** [review]
💯

**Comment by vardges-stripe** [comment]
@kgaidis-stripe thanks for the suggestion, made use of that.

**Pull request merged**